src/main/java/org/pac4j/vertx/auth/Pac4jUser.java

```diff
@@ -76,15 +76,15 @@ public void writeToBuffer(Buffer buff) {
 
     @Override
     public int readFromBuffer(int pos, Buffer buffer) {
-        pos = super.readFromBuffer(pos, buffer);
-        final int jsonByteCount = buffer.getInt(pos);
-        pos += 4;
-        final byte[] jsonBytes = buffer.getBytes(pos, pos + jsonByteCount);
-        pos += jsonByteCount;
+        int posLocal = super.readFromBuffer(pos, buffer);
```

Comment:
Am I missing something? It looks as though we're not properly moving posLocal on by the size of the count (4 bytes) or the number of bytes we read from the buffer to retrieve the stored json version of the underlying user profile. This would appear to me that we're going to return the wrong value.